### ISSUE TITLE
Frozen modules during SGRN fine-tuning

### ISSUE THREAD

**Issue by mandal4** [dohoseok/SGRN_torch#7]
Hi there,
I found that you freeze all the modules but set `requires_grad=True` for the following parameters. (line 356~365) in network_gcn.py)
new_cls_score_net.weight
new_cls_score_net.bias
new_bbox_pred_net.weight
new_bbox_pred_net.bias
relation_fc_1.weight
relation_fc_1.bias
relation_fc_2.weight
relation_fc_2.bias
gaussian.mu
gaussian.sigma
gaussian.bias
gaussian.lin.weight
sg_conv_1.weight
sg_conv_1.bias
sg_conv_2.weight
sg_conv_2.bias

However, when i loaded weight file for [Faster RCNN](https://drive.google.com/file/d/11k9WAGB5YmYwKDUu7kLW5E5q22AnOhzW/view?usp=sharing) and [SGRN](https://drive.google.com/file/d/1O2rGJqRXVlW9LVhwN5IvqDOLqWmYdPpP/view?usp=sharing) as you uploaded,
Other(following) parameters changed as well, which means they were set to 'requires_grad=True' during fine-tuning, exactly.
resnet.layer2.0.conv1.weight
resnet.layer2.0.conv2.weight
resnet.layer2.0.conv3.weight
resnet.layer2.0.downsample.0.weight
resnet.layer2.1.conv1.weight
resnet.layer2.1.conv2.weight
resnet.layer2.1.conv3.weight
resnet.layer2.2.conv1.weight
resnet.layer2.2.conv2.weight
resnet.layer2.2.conv3.weight
resnet.layer2.3.conv1.weight
resnet.layer2.3.conv2.weight
resnet.layer2.3.conv3.weight
resnet.layer3.0.conv1.weight
resnet.layer3.0.conv2.weight
resnet.layer3.0.conv3.weight
resnet.layer3.0.downsample.0.weight
resnet.layer3.1.conv1.weight
resnet.layer3.1.conv2.weight
resnet.layer3.1.conv3.weight
resnet.layer3.2.conv1.weight
resnet.layer3.2.conv2.weight
resnet.layer3.2.conv3.weight
resnet.layer3.3.conv1.weight
resnet.layer3.3.conv2.weight
resnet.layer3.3.conv3.weight
resnet.layer3.4.conv1.weight
resnet.layer3.4.conv2.weight
resnet.layer3.4.conv3.weight
resnet.layer3.5.conv1.weight
resnet.layer3.5.conv2.weight
resnet.layer3.5.conv3.weight
resnet.layer3.6.conv1.weight
resnet.layer3.6.conv2.weight
resnet.layer3.6.conv3.weight
resnet.layer3.7.conv1.weight
resnet.layer3.7.conv2.weight
resnet.layer3.7.conv3.weight
resnet.layer3.8.conv1.weight
resnet.layer3.8.conv2.weight
resnet.layer3.8.conv3.weight
resnet.layer3.9.conv1.weight
resnet.layer3.9.conv2.weight
resnet.layer3.9.conv3.weight
resnet.layer3.10.conv1.weight
resnet.layer3.10.conv2.weight
resnet.layer3.10.conv3.weight
resnet.layer3.11.conv1.weight
resnet.layer3.11.conv2.weight
resnet.layer3.11.conv3.weight
resnet.layer3.12.conv1.weight
resnet.layer3.12.conv2.weight
resnet.layer3.12.conv3.weight
resnet.layer3.13.conv1.weight
resnet.layer3.13.conv2.weight
resnet.layer3.13.conv3.weight
resnet.layer3.14.conv1.weight
resnet.layer3.14.conv2.weight
resnet.layer3.14.conv3.weight
resnet.layer3.15.conv1.weight
resnet.layer3.15.conv2.weight
resnet.layer3.15.conv3.weight
resnet.layer3.16.conv1.weight
resnet.layer3.16.conv2.weight
resnet.layer3.16.conv3.weight
resnet.layer3.17.conv1.weight
resnet.layer3.17.conv2.weight
resnet.layer3.17.conv3.weight
resnet.layer3.18.conv1.weight
resnet.layer3.18.conv2.weight
resnet.layer3.18.conv3.weight
resnet.layer3.19.conv1.weight
resnet.layer3.19.conv2.weight
resnet.layer3.19.conv3.weight
resnet.layer3.20.conv1.weight
resnet.layer3.20.conv2.weight
resnet.layer3.20.conv3.weight
resnet.layer3.21.conv1.weight
resnet.layer3.21.conv2.weight
resnet.layer3.21.conv3.weight
resnet.layer3.22.conv1.weight
resnet.layer3.22.conv2.weight
resnet.layer3.22.conv3.weight
resnet.layer4.0.conv1.weight
resnet.layer4.0.conv2.weight
resnet.layer4.0.conv3.weight
resnet.layer4.0.downsample.0.weight
resnet.layer4.1.conv1.weight
resnet.layer4.1.conv2.weight
resnet.layer4.1.conv3.weight
resnet.layer4.2.conv1.weight
resnet.layer4.2.conv2.weight
resnet.layer4.2.conv3.weight
rpn_net.weight
rpn_net.bias
rpn_cls_score_net.weight
rpn_cls_score_net.bias
rpn_bbox_pred_net.weight
rpn_bbox_pred_net.bias
cls_score_net.weight
cls_score_net.bias
bbox_pred_net.weight
bbox_pred_net.bias

In conclusion, i'd like to ask you **which parameter did you freeze during fine-tuning?**

**Comment by dohoseok** [comment]
I didn't edit the code for fine-tuning.
If the value is different, it might be my mistake.
Maybe, I trained SGRN from another weight file of which performance of Faster RCNN is saturated.

**Comment by mandal4** [comment]
Thanks for reply. 
And i found that parameters in ‘self.relation_fc_1’ and ‘self.relation_fc_2’ are not updated during fine-tuning phase for SGRN. 
I guess it occurs during ‘topk selection’  (line 442 in network_gcn.py)
Can you explain about it? 
Thanks,
